src/main/java/sg/ncl/ConnectionProperties.java

```diff
@@ -188,10 +188,26 @@ public String getRequest(String dataId, String requestId) {
         return HTTP_MODE + sioAddress + ":" + sioPort + "/" + dataEndpoint + "/" + dataId + "/requests/" + requestId;
     }
 
+    public String getPublicDataUsers() {
+        return HTTP_MODE + sioAddress + ":" + sioPort + "/" + dataEndpoint + "/public/users";
+    }
+
+    public String getPublicDataUser(String id) {
+        return HTTP_MODE + sioAddress + ":" + sioPort + "/" + dataEndpoint + "/public/users/" + id;
+    }
+
     public String getPublicData() {
         return HTTP_MODE + sioAddress + ":" + sioPort + "/" + dataEndpoint + "?visibility=PUBLIC";
     }
 
+    public String getPublicDataset(String dataId) {
+        return HTTP_MODE + sioAddress + ":" + sioPort + "/" + dataEndpoint + "/" + dataId + "?visibility=PUBLIC";
+    }
+
+    public String downloadPublicOpenResource(String dataId, String resourceId) {
+        return HTTP_MODE + sioAddress + ":" + sioPort + "/" + dataEndpoint + "/" + dataId + "/resources/" + resourceId + "/download?visibility=PUBLIC";
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "/resources/" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)